src/main/java/net/openhft/chronicle/network/cluster/ConnectionManager.java

```diff
@@ -64,4 +97,37 @@ public synchronized void onConnectionChanged(boolean isConnected, @NotNull final
          */
         void onConnectionChange(T nc, boolean isConnected);
     }
+
+    /**
+     * A connection listener that knows when it was added
+     */
+    private class ConnectionListenerHolder {
+        private final int sequence;
+        private final ConnectionListener<T> connectionListener;
+
+        public ConnectionListenerHolder(int sequence, ConnectionListener<T> connectionListener) {
+            this.sequence = sequence;
+            this.connectionListener = connectionListener;
+        }
+    }
+
+    /**
+     * An <strong>opaque</strong> token that NetworkContext connection event emitters need to retain and
+     * provide when they dispatch events.
+     * <p>
+     * Keeps track of the listeners they have executed and what the connected state was
+     * on their previous call.
+     * <p>
+     * Allows this state to immediately available, and be garbage collected with the emitter.
+     */
```

Comment:
This class can be final. This reduces the API commitment substantially as we now are free to update its contents with no risk of causing problems with potential classes extending this class.

Reply:
Good pickup